packages/client/src/ipfs/index.ts

```diff
@@ -3,34 +3,70 @@ import { IPFSClientConfig } from '../types';
 export default class IPFSClient {
   static readonly IPFS_CLIENT_ERROR = 'IPFS client not initialised';
   ipfs: any;
-  authorization: any;
+  authorization: string | undefined; 
+  provider: string | undefined;
+  initialized: boolean = false;
 
   constructor(ipfsClientConfig: IPFSClientConfig) {
-    const authorization =
-      'Basic ' + btoa(ipfsClientConfig.clientId + ':' + ipfsClientConfig.clientSecret);
-    this.authorization = authorization;
-    // ipfs-http-client is being mocked by src/__mocks__/ipfs-http-client
-    // due to the way its being imported in ipfs/index.ts, we have to add a special mock for it
-    import('ipfs-http-client').then(({ create }) => {
-      this.ipfs = create({
-        url: ipfsClientConfig.baseUrl,
-        headers: {
-          authorization,
-        },
+    this.provider = ipfsClientConfig.provider;
+    if (ipfsClientConfig.provider === 'quicknode') {
+      this.authorization = ipfsClientConfig.apiKey;
+      this.initialized = true;
+    }
+    else if (ipfsClientConfig.provider === 'infura') {
+      const authorization =
+        'Basic ' + btoa(ipfsClientConfig.clientId + ':' + ipfsClientConfig.clientSecret);
+      this.authorization = authorization;
+      import('kubo-rpc-client').then(({ create }) => {
+        this.ipfs = create({
+          url: ipfsClientConfig.baseUrl,
+          headers: {
+            authorization,
+          },
+        });
       });
-    });
+      this.initialized = true;
+    }
   }
 
   public async post(data: string): Promise<string> {
-    const formData = new FormData();
-    formData.append('file', new Blob([data], { type: 'application/json' }));
-
-    if (this.ipfs) {
-      const result = await this.ipfs.add(data);
-      await this.ipfs.pin.add(result.path);
-      return result.path;
+    if (!this.initialized) {
+      throw Error('IPFS client not initialised properly');
     }
 
-    throw Error('IPFS client not intiialised properly');
+    if (this.authorization) {
+      if (this.provider === 'quicknode') {
```

Comment:
instead of using if else-if, we can use switch case. That way adding more providers in the future will be simpler.